pkgs/tools/compression/dtrx/default.nix

```diff
@@ -0,0 +1,38 @@
+{ lib, fetchFromGitHub, python3Packages
+, gnutar, unzip, lhasa, rpm, binutils, cpio, gzip, p7zip, cabextract, unrar, unshield
+, bzip2, xz, lzip
+# unzip is handled by p7zip
+, unzipSupport ? false
+, unrarSupport ? false }:
+
+let
+  archivers = lib.makeBinPath ([ gnutar lhasa rpm binutils cpio gzip p7zip cabextract unshield ]
+  ++ lib.optional (unzipSupport) unzip
+  ++ lib.optional (unrarSupport) unrar
+  ++ [ bzip2 xz lzip ]);
+
+in python3Packages.buildPythonApplication rec {
+  pname = "dtrx";
+  version = "8.2.1";
+
+  src = fetchFromGitHub {
+    owner = "dtrx-py";
+    repo = "dtrx";
+    rev = version;
+    sha256 = "sha256-ayQ52teXWpw3ZvPhfqxFHxMZatQK9NKv/97ovANFZcE=";
+  };
+
+  postInstall = ''
+    wrapProgram "$out/bin/dtrx" --prefix PATH : "${archivers}"
+  '';
+
+  buildInputs = [ python3Packages.twine ];
+
+  meta = with lib; {
+    description = "Do The Right Extraction: A tool for taking the hassle out of extracting archives";
+    homepage = "https://github.com/dtrx-py/dtrx";
+    license = licenses.gpl3Plus;
+    maintainers = [ maintainers.spwhitt ];
+    platforms = platforms.all;
```

Comment:
Python packages do not need to have `meta.platforms` set. Indeed, Python is not necessarily built on all architectures.